### ISSUE TITLE
Plack::Request->content erroneously assumes it's working with seekable input

### ISSUE THREAD

**Issue by SineSwiper** [plack/Plack#642]
Noticing errors like this:

```
Can't locate object method "seek" via package "IO::Handle" at ..../lib/perl5/Plack/Request.pm line 78
```

The content method seems to think that it can seek psgi.input, even though Plack's own Plack::Handler::FCGI module does this:

```
    @{$self}{qw(stdin stdout stderr)}
      = (IO::Handle->new, IO::Handle->new, IO::Handle->new);

    my %env;
    my $request = FCGI::Request(
        $self->{stdin}, $self->{stdout}, $self->{stderr},
        \%env, $sock,
        ($self->{nointr} ? 0 : &FCGI::FAIL_ACCEPT_ON_INTR),
    );
```

And eventually puts that into psgi.input: `'psgi.input'        => $self->{stdin},`

Normally, `$self->{stdin}` would be a FCGI::Stream object after the `$request->Accept`, but there are conditions where that doesn't apply:

```
        if (!request->svout) {
            newSVrv(request->svout = newSV(0), "FCGI::Stream");
            newSVrv(request->sverr = newSV(0), "FCGI::Stream");
            newSVrv(request->svin = newSV(0), "FCGI::Stream");
        }
        sv_setiv(SvRV(request->svout), INT2PTR(IV, fcgx_req->out));
        sv_setiv(SvRV(request->sverr), INT2PTR(IV, fcgx_req->err));
        sv_setiv(SvRV(request->svin), INT2PTR(IV, fcgx_req->in));
        FCGI_Bind(request);
        request->accepted = TRUE;
```

Furthermore, FCGI::Stream doesn't even support seeking, so it wouldn't work, anyway:

```
sub SEEK {
    require Carp;
    Carp::croak(q/Operation 'SEEK' not supported on FCGI::Stream handle/);
}
```

**Comment by miyagawa** [comment]
I think this used to not be a problem, because Plack::Request checks if `psgix.input.buffered` is set in the environment, and otherwise slurps the input and replaces it with a temporary filehandle that does support `seek`.

Quick git search shows that it was changed in db1e817508d9ee08bc6df9474795e2767e7a9f0a

**Comment by miyagawa** [comment]
this should be fixed with #657
